README.md

```diff
@@ -29,122 +37,1058 @@ The library design is inspired by the work in [Exploring HTN Planners through ex
+const plusOneAction = plusOne({ target: 3 });
+
+// `zip` packs the action (including condition) and returns
+// a callable function
+const doPlusOne = zip(plusOneAction);
+
+// This executes the action. Useful for testing
+console.log(await doPlusOne(0)); // 1
+
+// plusTwoMethod is of type `Method`
+const plusTwoMethod = plusTwo({ target: 3 });
+
+// zip in this case expands the method into its actions so it can be applied
+// as if it was a function
+const doPlusTwo = zip(plusTwoMethod);
+doPlusTwo(0); // 2
+```
+
+Methods are useful for tweaking the plans under certain conditions. They also help reduce the search space. When looking for a plan, the Planner will try methods first, and only if methods fail, proceed to look for action tasks. During planning, the method is expanded recursively into its component actions, so they won't appear on the final plan.
+
+```typescript
+const planner = Planner.from({
+	tasks: [plusOne, plusTwo],
+});
+
+const res = planner.findPlan(0, 3);
+
+// The method has already been expanded
+expect(stringify(res)).to.deep.equal(sequence('+1', '+1', '+1'));
+```
+
+We can see method expansion by diagramming the planning process. Mahler provides a test util to generate [Mermaid](https://mermaid.js.org/) diagrams. We can pass it as a
+
+```typescript
+import { mermaid } from 'mahler/testing';
+
+// Create a tracer using the mermaid tool.
+const trace = mermaid();
+const planner = Planner.from({
+	tasks: [plusOne, plusTwo],
+	// Pass the trace as a configuration option to the planner
+	config: { trace },
+});
+
+// Find a plan from 0 to 3
+const res = planner.findPlan(0, 3);
+
+// render() returns a mermaid valid graph
+console.log(trace.render());
+```
+
+Passing the output to the [Mermaid Live Editor](https://mermaid.live) will produce the following diagram.
+
+```mermaid
+graph TD
+    start(( ))
+    start -.- d0{ }
+    d0 -.- f751444[["+2"]]
+    f751444 -.- b795d43("+1")
+    b795d43 -.- 4eb0bdd("+1")
+    4eb0bdd -.- d1{ }
+    d1 -.- 580ece0[["+2"]]
+    580ece0 -.- 580ece0-err[ ]
+    580ece0-err:::error
+    d1 -.- d634a71("+1")
+    d634a71 -.- stop(( ))
+    stop:::finish
+    classDef finish stroke:#000,fill:#000
+    start:::selected
+    start --> b795d43
+    b795d43:::selected
+    b795d43 --> 4eb0bdd
+    4eb0bdd:::selected
+    4eb0bdd --> d634a71
+    d634a71:::selected
+    d634a71 --> stop
+    classDef error stroke:#f00
+    classDef selected stroke:#0f0
+```
+
+In the diagram, the nodes connected via the dotted lines show the planning process, while nodes connected via the solid arrows show the resulting plan. The diagram shows how the planner tries first the `+2` method, which results into the expansion `('+1', '+1')`, a second tries fails, because the condition no longer holds, so the planner adds a `+1` action next, which allows the planner to reach the target. If we compare this diagram with one where the planner doesn't know have the `plusTwo` method, we get.
```

Comment:
```suggestion
In the diagram, the nodes connected via the dotted lines show the planning process, while nodes connected via the solid arrows show the resulting plan. The diagram shows how the planner tries first the `+2` method, which results into the expansion `('+1', '+1')`, a second try fails, because the condition no longer holds, so the planner adds a `+1` action next, which allows the planner to reach the target.

Let's compare this diagram with one where the planner doesn't know the `plusTwo` method. This new diagram looks like this:
```